### PR TITLE
Lookout fix URL params conflicting with local storage, enable annotations in URL params

### PR DESCRIPTION
**Pull request by carlocamurri**
Previously, column filters in the Jobs table were set from both local storage and URL parameters. This causes confusion when providing URL parameters if local storage is present already.

This change makes it so the filters are only inferred from the URL, whilst retaining local storage of which columns are displayed along with auto refresh logic.

Note that this change also enables annotation filters to be represented as URL params.

### CODE REVIEW

File: internal/lookout/ui/src/services/JobsQueryParamsService.ts [hunk 1]
```diff
@@ -98,13 +80,18 @@ export default class JobsQueryParamsService {
   }
 
   saveState(state: JobsContainerState) {
+    console.log(state.defaultColumns)
```
Comment (kannon92) [inline internal/lookout/ui/src/services/JobsQueryParamsService.ts:83]: Debug logging left over?

File: internal/lookout/ui/src/services/JobsLocalStorageService.ts [hunk 2]
```diff
@@ -9,36 +9,44 @@ export type JobsLocalStorageState = {
   annotationColumns?: ColumnSpec<string>[]
 }
 
-function convertToLocalStorageState(loadedData: Record<string, unknown>): JobsLocalStorageState {
+function isColumnArray(data: unknown): boolean {
```
Comment (kannon92) [inline internal/lookout/ui/src/services/JobsLocalStorageService.ts:12]: Any objection for adding some unit test for these functions?

File: internal/lookout/ui/src/services/JobsLocalStorageService.ts [hunk 3]
```diff
@@ -9,36 +9,44 @@ export type JobsLocalStorageState = {
...
+  return data !== undefined && Array.isArray(data) && data.every((col: any) => isColumnSpec(col))
+}
+
+function loadedDataIsValid(loadedData: Record<string, unknown>): boolean {
+  return (
+    loadedData.autoRefresh !== undefined &&
+    typeof loadedData.autoRefresh == "boolean" &&
+    isColumnArray(loadedData.defaultColumns) &&
```
Comment (kannon92) [inline internal/lookout/ui/src/services/JobsLocalStorageService.ts:20]: Are defaultColumns and annotationColumns always defined?

Comment (carlocamurri) [reply internal/lookout/ui/src/services/JobsLocalStorageService.ts:20]: No, the null check happens in `isColumnArray`

**Review by kannon92 — COMMENTED** [review]
I don't really have the context to understand the bug fix.  The code and tests look good to me.  

Could you explain how to test this and I'll see if I can grok the issue and see if this fixes it?

File: internal/lookout/ui/src/services/JobsLocalStorageService.ts (hunk 2 above)
Comment (kannon92) [reply internal/lookout/ui/src/services/JobsLocalStorageService.ts:12]: Thank you!

File: internal/lookout/ui/src/services/JobsLocalStorageService.ts [hunk 4]
```diff
@@ -9,36 +9,44 @@ export type JobsLocalStorageState = {
...
 
-function convertToLocalStorageState(loadedData: Record<string, unknown>): JobsLocalStorageState {
+function isColumnArray(data: unknown): boolean {
+  return data !== undefined && Array.isArray(data) && data.every((col: any) => isColumnSpec(col))
+}
+
+function loadedDataIsValid(loadedData: Record<string, unknown>): boolean {
+  return (
+    loadedData.autoRefresh !== undefined &&
+    typeof loadedData.autoRefresh == "boolean" &&
```
Comment (JamesMurkin) [inline internal/lookout/ui/src/services/JobsLocalStorageService.ts:19]:
```suggestion
    typeof loadedData.autoRefresh === "boolean" &&
```

**Review by JamesMurkin — APPROVED** [review]

**Pull request merged**